app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java

```diff
@@ -0,0 +1,133 @@
+package org.pesmypetcare.mypetcare.activities.fragments;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.material.button.MaterialButton;
+import com.google.android.material.datepicker.MaterialDatePicker;
+
+import org.pesmypetcare.mypetcare.R;
+import org.pesmypetcare.mypetcare.databinding.FragmentRegisterPetBinding;
+import org.pesmypetcare.mypetcare.features.pets.Pet;
+
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
+public class RegisterPetFragment extends Fragment {
+    private FragmentRegisterPetBinding binding;
+    private MaterialDatePicker materialDatePicker;
+    private RegisterPetCommunication communication;
+    private Button birthDate;
+    private boolean isBirthDateSelected;
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        binding = FragmentRegisterPetBinding.inflate(inflater, container, false);
+        communication = (RegisterPetCommunication) getActivity();
+        isBirthDateSelected = false;
+
+        setCalendarPicker();
+        setGenderDropdownMenu();
+        setUpAddNewPetButton();
+
+        return binding.getRoot();
+    }
+
+    /**
+     * Sets up the new pet button.
+     */
+    private void setUpAddNewPetButton() {
+        MaterialButton btnNewPet = binding.btnAddPet;
+        btnNewPet.setOnClickListener(v -> {
+            if (isAnyFieldBlank()) {
+                Toast errorMsg = Toast.makeText(getActivity(), R.string.error_empty_field, Toast.LENGTH_LONG);
+                errorMsg.show();
+            }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['}' at column 13 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)